src/StoreKeeper/WooCommerce/B2C/Cron/ProcessTaskCron.php

```diff
@@ -19,15 +19,26 @@ class ProcessTaskCron
      */
     public function execute(): void
     {
+        CronOptions::set(CronOptions::LAST_PRE_EXECUTION_DATE, date('Y-m-d H:i:s'));
+        $beforeCount = ProcessAllTasks::countTasks();
         try {
             $runner = Core::getCommandRunner();
-            $runner->execute(ProcessAllTasks::getCommandName(), [
-                'task_limit' => self::TASK_LIMIT,
+            $runner->execute(ProcessAllTasks::getCommandName(), [], [
+                'limit' => self::TASK_LIMIT,
             ]);
-            StoreKeeperOptions::set(StoreKeeperOptions::CRON_EXECUTION_LAST_STATUS, CronRegistrar::STATUS_SUCCESS);
+            CronOptions::set(CronOptions::LAST_EXECUTION_STATUS, CronRegistrar::STATUS_SUCCESS);
+            CronOptions::delete(CronOptions::LAST_POST_EXECUTION_ERROR);
         } catch (Throwable $throwable) {
-            StoreKeeperOptions::set(StoreKeeperOptions::CRON_EXECUTION_LAST_STATUS, CronRegistrar::STATUS_FAILED);
+            CronOptions::set(CronOptions::LAST_EXECUTION_STATUS, CronRegistrar::STATUS_FAILED);
+            CronOptions::set(CronOptions::LAST_POST_EXECUTION_ERROR, $throwable->getMessage());
             throw $throwable;
+        } finally {
+            $afterCount = ProcessAllTasks::countTasks();
```

Comment:
just check if there are any more tasks with status=new

Reply:
Changed to `countNewTasks`.